elastic/kibana
# [Search][Discover] Restore searchSessionId from URL

## Release note

Adds the Search Sessions support to Discover in Kibana

## Summary 

Build on top of https://github.com/elastic/kibana/pull/81297
Part of https://github.com/elastic/kibana/issues/61738
Similar merged pr for a dashboard app: https://github.com/elastic/kibana/pull/81489

## Summary

* Restore `searchSessionId` from URL
* Add `searchSessionId` support to discover's URL generator
* Pass `searchSessionId` into inspector

### Additional info

* Didn't do any additional handling to return to background search using back button. `searchSessionId` is removed from the URL on subsequent searches (e.g. after refresh or state change). Created a ticket, please see explanation of the problem: https://github.com/elastic/kibana/issues/82419

### How to test 

You can build a discover URL with fake session id: 

```
${discover}&searchSessionId=__fake__
```

Open it.
And you should see `__fake__` as session id in inspector.

^^ this is what a functional test does


### Checklist

Delete any items that are not applicable to this PR.
